scikeras/wrappers.py

```diff
@@ -516,39 +524,37 @@ def _check_array_dtype(arr):
                         f"`y` has {y_ndim_} dimensions, but this {self.__name__}"
                         f" is expecting {self.y_ndim_} dimensions in `y`."
                     )
-        X = check_array(X, allow_nd=True, dtype=_check_array_dtype(X))
-        X_dtype_ = X.dtype
-        X_shape_ = X.shape
-        n_features_in_ = X.shape[1]
-        if reset:
-            self.X_dtype_ = X_dtype_
-            self.X_shape_ = X_shape_
-            self.n_features_in_ = n_features_in_
-        else:
-            if not np.can_cast(X_dtype_, self.X_dtype_):
-                raise ValueError(
-                    f"Got `X` with dtype {X_dtype_},"
-                    f" but this {self.__name__} expected {self.X_dtype_}"
-                    f" and casting from {X_dtype_} to {self.X_dtype_} is not safe!"
-                )
-            if len(X_shape_) != len(self.X_shape_):
-                raise ValueError(
-                    f"`X` has {len(X_shape_)} dimensions, but this {self.__name__}"
-                    f" is expecting {len(self.X_shape_)} dimensions in `X`."
-                )
-            # The following check is a backport from
-            # sklearn.base.BaseEstimator._check_n_features
-            # since this method is not available in sklearn <= 0.22.0
-            if n_features_in_ != self.n_features_in_:
-                raise ValueError(
-                    "X has {} features, but {} is expecting {} features "
-                    "as input.".format(
-                        n_features_in_, self.__class__.__name__, self.n_features_in_
+        if X is not None:
```

Comment:
Why would `X` be None?

Reply:
Must be a leftover from a previous iteration; I couldn't find a single situation where this would be required. I'll revert it. Good catch, thank you.